### DISCUSSION TITLE
Feature Request: Support Pre/Post Scripts

### DISCUSSION THREAD

**Discussion by AssassinPrince** [darrenburns/posting#69] [Ideas]
As the title says it all, some defined scripts or commands available to run, `before` or `after` sending a request would be nice.
This was the shortest way to propose it, I'm sure you will have better usage ideas of this than what I already have in mind.

**Comment by darrenburns** [comment]
Thanks - it's on the road map!

**Comment by halostatue** [comment]
One use case in particular: I have written a specification and sample implementations (in Ruby, Node, and Elixir) for a very lightweight application authentication mechanism ([AppIdentity](https://github.com/kineticcafe/app_identity)) and the recommended versions all generate identity tokens with a relatively short lifetime (about ±10 minutes, but this is configurable and should probably be defaulted to shorter).

These tokens are generated from a locally stored identifier and secret and a timestamp-based nonce and passed in a custom header (the header's name is application-dependent) and should be regenerated on each request.

Any other time-sensitive HMAC will also benefit from this sort of functionality.

**Comment by darrenburns** [comment]
Forgot to update this discussion, but this was implemented a while back :)
